### PR TITLE
ref: f-stringify tests/

### PR DESCRIPTION
**Pull request by joshuarli**
First piece of https://github.com/getsentry/sentry/pull/23436, where it was decided that single classes of fixes should be applied one at a time, starting with the entirety of `tests/`.

This is using [my own fork of pyupgrade](https://github.com/joshuarli/pyupgrade/tree/joshuarli/individual-fixers) which has some terrible patches to selectively apply some classes of fixes.

I should note, this is relatively conservative, pyupgrade won't attempt to rewrite more complicated `format`s (https://github.com/asottile/pyupgrade/issues/38). I made a few changes to do the following:

- don't bail on multiline
- inline string literals

### CODE REVIEW

File: tests/apidocs/endpoints/organizations/test-org-users.py [hunk 1]
```diff
@@ -1,6 +1,3 @@
-# -*- coding: utf-8 -*-
```
Comment (EvanPurkhiser) [inline tests/apidocs/endpoints/organizations/test-org-users.py:1]: Could we just remove these everywhere first in a diff PR?

Comment (joshuarli) [reply tests/apidocs/endpoints/organizations/test-org-users.py:1]: Sure.

**Comment by joshuarli** [comment]
Hmm, I knew that it was conservative, but trivial things like this aren't being changed:

```
        resp = self.client.get(
            "{}?redirect_uri={}&client_id={}".format(
                self.path, "https://example.com", self.application.client_id
            )
        )
```

Gonna try to get this working.

**Comment by joshuarli** [comment]
Still a good amount of formats, this one's not being fixed: `title = "{}: {}".format("Resolved", alert_rule.name)`. Hmm.

File: tests/sentry/eventstream/kafka/test_consumer.py [hunk 2]
```diff
@@ -458,9 +458,9 @@ def on_assign(consumer, assignment):
             # Move the committed offset forward for our synchronizing group.
             producer.produce(
                 commit_log_topic,
-                key="{}:{}:{}".format(
-                    expected_message.topic(), expected_message.partition(), synchronize_commit_group
-                ).encode("utf8"),
+                key=f"{expected_message.topic()}:{expected_message.partition()}:{synchronize_commit_group}".encode(
+                    "utf8"
```
Comment (joshuarli) [inline tests/sentry/eventstream/kafka/test_consumer.py:462]: next up: removing encode where appropriate lol

**Comment by joshuarli** [comment]
Figured out how to selectively fix printf-style formatting, the remaining and hardest thing is to allow inlining of string literals.

**Comment by joshuarli** [comment]
Oh, managed to get string literal inlining working. There's still... more format calls than I'd like to manually audit. I think this is good enough, the remaining isn't really worth the time it'd take.

**Comment by joshuarli** [comment]
I'm gonna redo this PR, now that I think my fork is in a good place.

**Pull request closed without merging**